core/src/main/java/org/jdbi/v3/core/mapper/reflect/ConstructorMapper.java

```diff
@@ -207,14 +207,14 @@ private Optional<RowMapper<T>> createSpecializedRowMapper(StatementContext ctx,
                                                List<String> unmatchedColumns) {
         final int count = factory.getParameterCount();
         final Parameter[] parameters = factory.getParameters();
-        final Type[] types = factory.getTypes();
+        final List<Type> types = factory.getTypes();
```

Comment:
Had to convert to list due to error about exposing an array. Thinking to change the api to store the array internally and change the signature to `getType(index)`, wdyt?

Reply:
That sounds ok